src/main/webapp/app/course/manage/course-management.service.ts

```diff
@@ -160,11 +167,22 @@ export class CourseManagementService {
     }
 
     /**
-     * finds one course using a GET request
+     * Finds one course using a GET request.
+     * If the course was already loaded it can be retrieved using {@link CourseStorageService#getCourse} or {@link CourseStorageService#subscribeToCourseUpdates}
      * @param courseId the course to fetch
      * @param userRefresh whether this is a user-initiated refresh (default: false)
      */
     findOneForDashboard(courseId: number, userRefresh = false): Observable<EntityResponseType> {
+        if (this.inProduction === undefined) {
+            this.profileService.getProfileInfo().subscribe((profileInfo: ProfileInfo) => {
+                this.inProduction = profileInfo.inProduction;
+            });
+        }
+
+        if (!this.inProduction && this.courseStorageService.getCourse(courseId)) {
```

Comment:
why do we need this distinction for production vs. not production? I don't think we should do that

Reply:
@MaximilianAnzinger suggested this, so developers are notified whenever they use the rest call instead of the cached course. Otherwise another new rest call can easily sneak in in some future PR.

Reply:
I don't think this is a good idea, our tests should verify this and not some production code. I would kindly ask you to remove the code and to write a test that makes sure that no REST call sneaks in

Reply:
Removed 👍